athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java

```diff
@@ -187,44 +182,25 @@ public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest getTabl
                     totalRecordCount = rs.getLong(1);
                 }
                 if (totalRecordCount > 0) {
-                    // if number of partitions are more than defined limit "MAX_PARTITION_COUNT"
-                    // it will do maximum 50 partitions,49 partitions will have 500000 records each and last partition will have the remaining number of records.
-                    double limitValue = totalRecordCount / PARTITION_RECORD_COUNT;
+                    long partitionRecordCount = (long) (Math.ceil(totalRecordCount / MAX_PARTITION_COUNT));
+                    LOGGER.info(" Total Page  Count" +  partitionRecordCount);
+                    double limitValue = totalRecordCount / partitionRecordCount;
                     double limit = (int) Math.ceil(limitValue);
```

Comment:
What is the purpose of `limit` and `limitValue`? Wouldn't limit always be `MAX_PARTITION_COUNT(or 50)`

Reply:
Yes Akuzin,you are right.For most of the scenarios its 50 i.e. MAX_PARTITION_COUNT.
But for some cases,like when the records are less than 50,limit will not be equal to MAX_PARTITION_COUNT.
We discussed this point internally also and found that its not good to create unnecessary splits even for smaller number of records.
We were thinking to update the logic,if the number of records are less than or equal to 10000,it should be returned in a single split.
We need your opinion on this.

long partitionRecordCount = (long) (Math.ceil(totalRecordCount / MAX_PARTITION_COUNT));
double limitValue = (partitionRecordCount <= 200) ? (long) totalRecordCount : partitionRecordCount;
double limit = (int) Math.ceil(totalRecordCount / limitValue);

Reply:
Yes I agree, I think that would be the correct approach. Thank you